hook-worker/src/worker.rs

```diff
@@ -666,4 +682,29 @@ mod tests {
             ));
         }
     }
+
+    #[tokio::test]
+    async fn test_private_ips_denied() {
+        let method = HttpMethod::POST;
+        let url = "http://localhost:18081/echo";
+        let headers = collections::HashMap::new();
+        let body = "a very relevant request body";
+        let filtering_client =
+            build_http_client(Duration::from_secs(1), false).expect("failed to create client");
+
+        let err = send_webhook(filtering_client, &method, url, &headers, body.to_owned())
+            .await
+            .err()
+            .expect("request didn't fail when it should have failed");
+
+        assert!(matches!(err, WebhookError::Request(..)));
+        if let WebhookError::Request(request_error) = err {
```

Comment:
Is this being converted to a `Retryable` error here? https://github.com/rusty-hook/blob/4e0a5469b365a1e0a3e63348c0a08ed46ca8b9f5/hook-worker/src/worker.rs#L404-L409

I think something in here should assert this is a `WebhookRequestError::NonRetryableRetryableRequestError`?

Reply:
I left it as retryable in case there's a "good" reason for DNS to flips, but we could indeed make it non retryable easily. Would you prefer this?

Reply:
I can't make a good case for it being retryable, can you? I think either way we should probably assert one or the other just to document intention (and maybe say why, if it's retryable).

Reply:
Pushed a commit to not retry these